### PR TITLE
15538 Fixed court order type not being restored

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#15538

*Description of changes:*
- app version = 5.11.21
- added court order initial case
- fixed registrar initial case
- fixed lint warning

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-736-9p7t2cem.web.app

File: src/components/Restoration/ApprovalType.vue [hunk 1]
```diff
@@ -3,6 +3,7 @@
     <ApprovalTypeShared
       class="pa-8"
       :courtOrderNumber="getRestoration.courtOrder.fileNumber"
+      :approvedByCourtOrder="approvedByCourtOrder"
```
Comment (severinbeauvais) [inline src/components/Restoration/ApprovalType.vue:6]: This is needed to set the initial state of the (shared) ApprovalType component. Without this, the court order option (and its value) were not being restored/displayed.

File: src/components/Restoration/ApprovalType.vue [hunk 2]
```diff
@@ -44,8 +45,12 @@ export default class ApprovalType extends Vue {
   @Action(useStore) setRestorationCourtOrder!: (x: CourtOrderIF) => void
   @Action(useStore) setRestorationNoticeDate!: (x: string) => void
 
+  get approvedByCourtOrder (): boolean {
+    return !!this.getRestoration.courtOrder.fileNumber
+  }
+
   get approvedByRegistrar (): boolean {
-    return (!!this.getRestoration.noticeDate && !!this.getRestoration.applicationDate)
+    return (!!this.getRestoration.noticeDate || !!this.getRestoration.applicationDate)
```
Comment (severinbeauvais) [inline src/components/Restoration/ApprovalType.vue:53]: Bugfix: show this option if either date was entered.

**Review by ArwenQin — APPROVED** [review]
Looks good!

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by ketaki-deodhar — APPROVED** [review]

**Pull request merged**